plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -368,6 +369,22 @@ protected void verifyColumnName(DatabaseMetaData databaseMetadata, String column
         }
     }
 
+    @Override
+    public OptionalInt getMaxColumnNameLength(ConnectorSession session)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            int maxColumnNameLength = connection.getMetaData().getMaxColumnNameLength();
```

Comment:
Is this running a query in SqlServer? How often are we going to call this method? Same for other connectors.

Reply:
See https://github.com/trinodb/trino/pull/18984#discussion_r1322421663.

For Postgres for example it runs a query only once per instance of `Connection` since the driver caches the information. Oracle does the same.

SQL Server driver doesn't talk to the database at all for this method.

Reply:
this is now cached on the jdbc client and is only called once per query.

Note that for most databases this doesn't actually involve any remote interaction at all except the first time (for next times the JDBC driver itself caches the relevant information).